driver-core/src/main/java/com/datastax/driver/core/Token.java

```diff
@@ -671,4 +679,117 @@ public String toString() {
       return value.toString();
     }
   }
+
+  // CDCPartitioner tokens
+  // https://github.com/scylladb/scylla/blob/master/cdc/cdc_partitioner.cc
+  static class CDCToken extends TokenLong64 {
+    private final long value;
+
+    public static final Factory FACTORY = new CDCTokenFactory();
+
+    private static final int CDC_PARTITION_KEY_LENGTH = 16;
+    private static final long VERSION_MASK = 0xF;
+    private static final int MIN_SUPPORTED_VERSION = 1;
+    private static final int MAX_SUPPORTED_VERSION = 1;
+
+    private static class CDCTokenFactory extends Factory {
+      private static final Logger logger = LoggerFactory.getLogger(CDCTokenFactory.class);
+
+      private static final BigInteger RING_END = BigInteger.valueOf(Long.MAX_VALUE);
+      private static final BigInteger RING_LENGTH =
+          RING_END.subtract(BigInteger.valueOf(Long.MIN_VALUE));
+      static final CDCToken MIN_TOKEN = new CDCToken(Long.MIN_VALUE);
+      static final CDCToken MAX_TOKEN = new CDCToken(Long.MAX_VALUE);
+
+      @Override
+      CDCToken fromString(String tokenStr) {
+        return new CDCToken(Long.parseLong(tokenStr));
+      }
+
+      @Override
+      DataType getTokenType() {
+        return DataType.bigint();
+      }
+
+      @Override
+      Token deserialize(ByteBuffer buffer, ProtocolVersion protocolVersion) {
+        return new CDCToken(TypeCodec.bigint().deserialize(buffer, protocolVersion));
+      }
+
+      @Override
+      Token minToken() {
+        return MIN_TOKEN;
+      }
+
+      @Override
+      CDCToken hash(ByteBuffer partitionKey) {
+        int offset = partitionKey.position();
+        int length = partitionKey.remaining();
+
+        if (length != CDC_PARTITION_KEY_LENGTH) {
+          logger.warn(
+              "CDC partition key has invalid length: expected {} bytes, but got {} bytes",
+              CDC_PARTITION_KEY_LENGTH,
+              length);
+          return MIN_TOKEN;
```

Comment:
This may cause a disaster because all keys will end up with the same token. It would be better to take the first 8 bytes of the key if they are present. If length < 8 then add 0 to the front. It's important that key -> token mapping was stable otherwise the user won't be able to find the data so random token won't work.

Reply:
It is now implemented the same way as in Scylla:

https://github.com/scylladb/scylla/blob/a81b731a1a2f5a07b0b5b8fde7939be005be0a97/cdc/cdc_partitioner.cc#L42-L48

However I see the issue and I will fix it.

Reply:
Fair point. It might be worth fixing there as well.

Reply:
Fixed in https://github.com/scylladb/java-driver/pull/35/commits/68408434963e1ba34183c397e5d198558ed489bc